e2e_tests/tests/test_sdk.py

```diff
@@ -474,3 +474,32 @@ def test_rp_workspace_mapping() -> None:
     finally:
         for wid in workspace_ids:
             bindings.delete_DeleteWorkspace(session=sess, id=wid)
+
+
+@pytest.mark.e2e_cpu
+def test_experiment_w_template() -> None:
```

Comment:
We're less good with this than we are with our unit tests, but could you please rename the test to [describe the specific behavior being tested](https://hpe-aiatscale.atlassian.net/wiki/spaces/ENGINEERIN/pages/1351909582/Python+Testing#Best-practices)? What aspect of template use does this test target?

Reply:
My reading of this exact test is:

1. When creating an experiment with a minimal template, a trial is started before pytest times this test out
2. very soon after the first trial is started, the executed experiment has exactly one trial

I think I'd like a short chat about what exactly we should test. Should we test that the system can start an experiment with any template? Should we test that it actually uses config from the template?

I don't know! But I'd like to work out what to test and what assertions and names those tests should have.

Reply:
Added some explanation and add test for "actually uses config from the template"

Reply:
I like it!